proposals/0263-System-Request.md

```diff
@@ -64,30 +70,82 @@ Here are the changes needed by SDL Core.
-
-- When SDL Core revives a SystemRequest response or OnSystemRequest notification from HMI, if the optional `data`/`resultData` parameter exists, SDL Core shall ignore the `fileName` parameter. Instead of reading data from the file, SDL Core shall base64 decode the string, and send the resulting data in hybrid part of the message to mobile. If the optional `data`/`resultData` parameter does not exist, SDL Core reads data from file as before.
+- When SDL Core receives a SystemRequest request from mobile, if the incoming binary data is less than a certain size specified by smartdevicelink.ini, instead of saving the data to a temporary file, SDL Core shall encode the data using base64 format, and send the encoded string as `requestData` directly within a SystemRequest HMI message. If the size of data is large, SDL Core saves the data to a file and passes the file name to HMI as before.
 
-Potential changes to HMI
+- When SDL Core revives an OnSystemRequest notification from HMI, if the optional `data` parameter exists, SDL Core shall ignore the `fileName` parameter. Instead of reading data from the file, SDL Core shall base64 decode the string, and send the resulting data in hybrid part of the message to mobile. If the optional `data` parameter does not exist, SDL Core reads data from file as before.
 
-- When HMI receives a SystemRequest request, if optional parameter `requestData` exists, it shall ignore the `fileName` parameter. 
-- When HMI sends a SystemRequest response or a OnSystemRequest notification, it is free to choose either the file transfer (as before) or string transfer (new) regardless of the configurable parameter in smartdevicelink.ini.
+- When SDL Core revives a SystemRequest response from HMI, if the optional `resultData` parameter exists, SDL Core shall base64 decode the string, and send the resulting data in hybrid part of the message to mobile. 
 
-Potential changes to the mobile proxy lib
+Potential changes to HMI
 
-- Extract hybrid part of the message from SystemRequest RPC response message, if the library did not do that before.
+- When HMI receives a SystemRequest request, if optional parameter `requestData` exists, it shall ignore the `fileName` parameter. It shall base64 decode the string and use the data as if it comes from a file.
+
+- When HMI sends a OnSystemRequest notification, it is free to choose either the file transfer (as before) or string transfer (new) regardless of the configurable parameter in smartdevicelink.ini.
+
+- When HMI sends a SystemRequest response with `resultData`, it shall choose string transfer.
+
+Since `fileName` is mandatory in the HMI API for SystemRequest request and OnSystemRequest and it will be ignored when `requestData`/`data` exists, `fileName` can be hard coded to any string. This is true for both SDL core and HMI.
```

Comment:
```suggestion
Since `fileName` is mandatory in the HMI API for `SystemRequest` request and `OnSystemRequest` and it will be ignored when `requestData`/`data` exists, `fileName` can be hard coded to any string. This is true for both SDL Core and HMI.
```